### PR TITLE
[FLAT-217] live encode

### PR DESCRIPTION
**Pull request by pcorless**
Addition of separate build scripts for the openjpeg library and wasm builds.  Jenkins file is auto picked up and artifacts are uploaded to the morpheus-builds/openjpeg folder on a successful build replacing any previous artifact that has the same release version.   

RELEASE file needs to be updated for a given release with the current release version.  Artifacts are saved with the release version in the file name.   

For convenience there is also a build.sh which can be used to build the docker images and artifacts locally.

### CODE REVIEW

File: build-wasm.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        -s ERROR_ON_MISSING_LIBRARIES=1 \
+        -s USE_PTHREADS=1 \
+        -s PTHREAD_POOL_SIZE=8 \
+        -s TOTAL_MEMORY=256MB \
+        -s ENVIRONMENT="web,worker" \
+        -s EXTRA_EXPORTED_RUNTIME_METHODS="['ccall', 'writeArrayToMemory', 'getValue']" \
+        -s EXPORTED_FUNCTIONS="['_jp2_decode']"
+)
+echo "============================================="
+echo "Compiling openjpeg wasmdone"
```
Comment (tjugdev) [inline build-wasm.sh]: Missing space

File: build-wasm.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ENVIRONMENT="production";
+BROWSER_OUTPUT="openjpeg"
+OPTIMIZE="-Os"
+if [$ENVIRONMENT != "production"]
+then
+    OPTIMIZE="-O2 --js-opts 0 -g4 --source-map-base http://localhost:8080/"
+fi
+LDFLAGS="${OPTIMIZE}"
+CFLAGS="${OPTIMIZE}"
+CXXFLAGS="${OPTIMIZE}"
```
Comment (tjugdev) [inline build-wasm.sh]: Do these flags actually do anything? We use `$OPTIMIZE` directly later

File: build-wasm.sh [hunk 3]
```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+RELEASE_VERSION=$(cat RELEASE)
+DIR=$(cd $(dirname ${BASH_SOURCE[0]}) && pwd)
+BUILD_DIR="${DIR}/build-wasm"
+DIST_DIR="${DIR}/dist/openjpeg-wasm"
+
+ENVIRONMENT="production";
```
Comment (tjugdev) [inline build-wasm.sh]: Should this be overridable? Something like `ENVIRONMENT=${ENVIRONMENT:-production}`?

File: build-wasm.sh [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+BROWSER_OUTPUT="openjpeg"
+OPTIMIZE="-Os"
+if [$ENVIRONMENT != "production"]
+then
+    OPTIMIZE="-O2 --js-opts 0 -g4 --source-map-base http://localhost:8080/"
```
Comment (tjugdev) [inline build-wasm.sh]: We should remove the `source-map-base` flag here since I'm pretty sure it doesn't work.

File: build-wasm.sh (hunk 2 above)
Comment (pcorless) [reply build-wasm.sh]: Yeah, they can be cut now, they've been there so long now I don't even see them.

**Pull request merged**